pysat/tests/test_instruments.py

```diff
@@ -103,278 +100,115 @@ def init_func_external(self):
+        module = import_module(''.join(('.', name)),
+                               package=self.package)
         info = module._test_dates
-        check = []
         for sat_id in info.keys():
             for tag in info[sat_id].keys():
-                check.append(isinstance(info[sat_id][tag], pysat.datetime))
-        assert np.all(check)
+                assert isinstance(info[sat_id][tag], pds.datetime)
 
-    def check_download(self, inst):
-        from unittest.case import SkipTest
-        import os
+class TestInstrumentsDownload():
 
+    @pytest.mark.first
+    @pytest.mark.parametrize("inst", instruments['list'])
+    def test_download(self, inst):
+        print(' '.join(('\nChecking download routine functionality for module: ',
+                        inst.platform, inst.name, inst.tag, inst.sat_id)))
         start = inst._test_dates[inst.sat_id][inst.tag]
-        try:
-            # check for username
-            inst_name = '_'.join((inst.platform, inst.name))
-            if inst_name in user_download_dict:
-                inst.download(start, start,
-                              user=user_download_dict[inst_name][0],
-                              password=user_download_dict[inst_name][1])
-            else:
-                inst.download(start, start)
-        except Exception as e:
-            # couldn't run download, try to find test data instead
-            print("Couldn't download data, trying to find test data.")
-            saved_path = pysat.data_dir
-
-            new_path = os.path.join(pysat.__path__[0], 'tests', 'test_data')
-            pysat.utils.set_data_dir(new_path, store=False)
-            _test_dates = inst._test_dates
-            inst = pysat.Instrument(platform=inst.platform,
-                                    name=inst.name,
-                                    tag=inst.tag,
-                                    sat_id=inst.sat_id,
-                                    temporary_file_list=True)
-            inst._test_dates = _test_dates
-            pysat.utils.set_data_dir(saved_path, store=False)
-            if len(inst.files.files) > 0:
-                print("Found test data.")
-                raise SkipTest
+        # check for username
+        inst_name = '_'.join((inst.platform, inst.name))
+        dl_dict = user_download_dict[inst_name] if inst_name in \
+            user_download_dict.keys() else {}
+        inst.download(start, start, **dl_dict)
+        assert len(inst.files.files) > 0
+
+    @pytest.mark.second
+    @pytest.mark.parametrize("inst", instruments['list'])
+    @pytest.mark.parametrize("clean_level", ['none', 'dirty', 'dusty',
+                                             'clean'])
+    def test_load(self, inst, clean_level):
+        # make sure download was successful
+        print(' '.join(('\nChecking load routine functionality for module',
+                        inst.platform, inst.name, inst.tag, inst.sat_id)))
```

Comment:
Again, as long as the asserts tell us which values fail, I don't see the harm in the more cryptic statements.  However, we already have a log that's too long to read in the default Travis view.

Reply:
If I give an instrument the wrong web address for the download routine, here is the output:
```
_______________________________________________ TestInstrumentsDownload.test_download[inst0] ________________________________________________

self = <pysat.tests.test_instruments.TestInstrumentsDownload object at 0x1a2513f320>
inst = <pysat._instrument.Instrument object at 0x1a2507c278>

    @pytest.mark.first
    @pytest.mark.parametrize("inst", instruments['list'])
    def test_download(self, inst):
        print(' '.join(('\nChecking download routine functionality for module: ',
                        inst.platform, inst.name, inst.tag, inst.sat_id)))
        start = inst._test_dates[inst.sat_id][inst.tag]
        # check for username
        inst_name = '_'.join((inst.platform, inst.name))
        dl_dict = user_download_dict[inst_name] if inst_name in \
            user_download_dict.keys() else {}
        inst.download(start, start, **dl_dict)
>       assert len(inst.files.files) > 0
E       assert 0 > 0
E        +  where 0 = len(Series([], dtype: object))
E        +    where Series([], dtype: object) = <pysat._files.Files object at 0x1a2507ca20>.files
E        +      where <pysat._files.Files object at 0x1a2507ca20> = <pysat._instrument.Instrument object at 0x1a2507c278>.files

pysat/tests/test_instruments.py:183: AssertionError
```
The travis log has been shortened quite a bit.  Part of this is the implementation of logging, part of this is how pytest handles warnings (summary at the end rather than each individual statement).  It still is incredibly long, but I can at least scroll to the end without going to the raw view.
https://travis-ci.org/github/pysat/pysat/jobs/661774525?utm_medium=notification

Reply:
Similar output for forcing the load routine to fail:
```
_______________________________________________ TestInstrumentsDownload.test_load[none-inst0] _______________________________________________

self = <pysat.tests.test_instruments.TestInstrumentsDownload object at 0x1a1fbf2a58>
inst = <pysat._instrument.Instrument object at 0x1a1fb43400>, clean_level = 'none'

    @pytest.mark.second
    @pytest.mark.parametrize("inst", instruments['list'])
    @pytest.mark.parametrize("clean_level", ['none', 'dirty', 'dusty',
                                             'clean'])
    def test_load(self, inst, clean_level):
        # make sure download was successful
        print(' '.join(('\nChecking load routine functionality for module',
                        inst.platform, inst.name, inst.tag, inst.sat_id)))
        if len(inst.files.files) > 0:
            inst.clean_level = clean_level
            target = 'Fake Data to be cleared'
            inst.data = [target]
            start = inst._test_dates[inst.sat_id][inst.tag]
            inst.load(date=start+pds.DateOffset(days=2))
            # Make sure fake data is cleared
            assert target not in inst.data
            # If cleaning not used, something should be in the file
            # Not used for other levels since cleaning may remove all data
            if clean_level == "none":
>               assert not inst.empty
E               assert not True
E                +  where True = <pysat._instrument.Instrument object at 0x1a1fb43400>.empty

pysat/tests/test_instruments.py:204: AssertionError
```

There's probably a way we could write this without passing the objects through as a parametrized list that would do what we want, but I'm not sure what that is.

Reply:
With the print statements, we can back which instrument is failing.
```
pysat/tests/test_instruments.py::TestInstrumentsDownload::test_download[inst0] 
Checking download routine functionality for module:  iss fpmu  
PASSED
pysat/tests/test_instruments.py::TestInstrumentsDownload::test_load[none-inst0] 
Checking load routine functionality for module iss fpmu  
FAILED
pysat/tests/test_instruments.py::TestInstrumentsDownload::test_load[dirty-inst0] 
Checking load routine functionality for module iss fpmu  
PASSED
```

Reply:
I think you could potentially make a decorator that wraps the whole method with a try except. Upon assertion error you could print values from the Instrument. I haven't actually made a decorator though so that is really just a guess. prints are ok with me in the meantime.